### DISCUSSION TITLE
$doc/window-close-if.js for certain sites?

### DISCUSSION THREAD

**Discussion by ghost** [uBlockOrigin/uAssets#17800] [Ideas]
Could `$doc` and/or `window-close-if.js` be used for betting / nsfw cam sites, on the same grounds [a scriptlet was added for `norton.com`](https://github.com/uBlockOrigin/uAssets/blob/ebb8057f3c1df20807612415235065931d68e3d3/filters/filters-2022.txt#L1374-L1375)?. I have found that betting sites tend to be used in popunder campgains.

To encounter such situations (with affiliate / promo params), I have ideas:

1. Visiting popunder-containing sites without an ad blocker (at own risk; I do not recommend this)
2. Using sites which offer redirection to such URLs

After visiting `https://siteuripariuri.ro/` and clicking on some links, I have come up with the following filters (for experimentation):

```
welcome.unibet.ro,promo.lasvegas.ro,getsbet.ro##+js(window-close-if, btag=)
sportpromo.mrbit.ro##+js(window-close-if, ?ref=)
lp.mozzartbet.ro##+js(window-close-if, /bonus)
```

Also, in my opinion, such things on Romanian betting sites could be addressed in uAssets, as popunders can appear on non-Romanian sites.

**Comment by mapx-** [comment]
I visited `https://siteuripariuri.ro`
I get no popup or popunder clicking `Activează bonus` , just normal navigation

**Reply by ghost** [reply]
That's not the point; the point is that some popunders may circumvent uBO, and so the visitor may be connected to some random site. I've found that betting sites tend to be used as destinations (as well as other affiliate programs).. If we can find a way to prevent the annoyance and privacy exposure that someone users have not expressed full consent to, I think that's a good thing

**Reply by mapx-** [reply]
When you'll find real cases post them here.

**Reply by ghost** [reply]
I can, but I don't want to post params that could dox me

an idea: visiting popunder-containing sites (you know of such sites), and yeah.. again, not recommended.

I'll see on my phone on Brave

**Comment by ghost** [comment]
From NSFW `sxyprn. com` on Brave (iOS) :

<details>

NSFW: `https://stripchat.com/barbiehumpxxx?campaignId=2ec5eb55f38cc54b34cd6e63ea23780457b898229dd2fd5056127d5ae881331e&p1=2494186&realDomain=go.xlivrdr.com&sound=off&sourceId=1941842&stripbotVariation=Empty`


`https://youbbster.com/edc/general/ro/qwe/lp31_tr_uwhnad/?clickid=GvzOA8U1&trackingid=e2143175mgxx91da&affclickid=12bb7p2ft8phqe57&fat=WIOAOtkSBKKbibOrhhZ8AdOc1B6tOnyqEzdKxM7DV5gpsOqWYxKFkCnzjAtHYU8VLzRnBV953ec4Sw14B6g0NDy3ZFpl%2BuQ3VeFgfo2K6bJ5zmAqLp5dNEy5fXQaU%2BtT&sub1=3&sub4=Unknown&sub5=Unknown` https://user-images.githubusercontent.com/131808021/234679744-d640fb1d-75d9-4883-ab37-3ba8df7cb5d2.png

`https://www.user-shield.com/stream-secure-video-stream-win1/?pid=adcash_int&c=324753420_23583118&af_siteid=6388140-808726207-0&af_click_lookback=7d&clickid=168253669409990TROTV4848371544V5cd1&advertiser=105233` (the page refreshed after I pressed on the button, idk if it changed the URL) https://user-images.githubusercontent.com/131808021/234680560-89f6cc31-e844-45ca-bb58-a6c3bccb030f.png

`https://mbest.aliexpress.com/m/promote.htm?tracelog=wwwhome2mobilesitehome&browser_redirect=true` (seemingly refreshed/redirected at some point)

`https://www.888casino.ro/sem/double-bonus.htm?utm_campaign=100138643_1853316_nodescription&utm_content=100138643&utm_medium=casap&utm_source=aff`

I think this is NSFW: `http://xhqxmovies.com/?id=mycjnet.com#`

</details>

**Comment by Yuki2718** [comment]
Generally, we do not block any sites by their contents except for malware or scam. We know betting sites tend to be destination of popunder, but blocking just because the site is betting, or tends to be destination of popunder, is unfair and breach of trust. If a site has unblocked popunder, that should be addressed for the site.

**Reply by ghost** [reply]
I was talking about blocking by affiliate parameters / subdomains, if we can identify such things

**Reply by Yuki2718** [reply]
We already block many of redirector by parameter if they lead to malicious sites.
